### DISCUSSION TITLE
used `headers().get('X-NEXT-INTL-LOCALE')`. `headers()` should be awaited before using its value.

### DISCUSSION THREAD

**Discussion by l-you** [amannn/next-intl#1525] [Q&A]
I get an error `used `headers().get('X-NEXT-INTL-LOCALE')`. `headers()` should be awaited before using its value.` after upgrading to next.js 15 . I use i18n routing with next-intl.
How to workaround it?

**Comment by amannn** [comment]
Please see: [next-intl 3.22: Incrementally moving forward](https://next-intl-docs.vercel.app/blog/next-intl-3-22)
